Mod2/Labs/Week2/TDD.md

```diff
@@ -0,0 +1,41 @@
+## Lab Objectives
+- Practice class brainstorming
+- Use TDD to implement Library classes
+
+### Brainstorm Classes
+
+In small groups, review the requirements below.  Brainstorm the classes and methods that we would need to implement the functionality.
+
+```
+Library
+
+We need to be able to create a Library that has books and patrons.
+
+A Patron should have a name and phone number.
+
+A Book should have a title, author, and a Patron (the patron will be null if the book is not checked out).
+
+The library should be able to tell us which books a given patron has checked out.
+
+The library should be able to tell us which books are NOT checked out.
+```
+
+Be ready to share your brainstorm!  As a cohort, we will develop one class diagram that we will use to build our tests.
+
+### Build Tests
+
+**Individually**
+1. Create a new Console Application called `Library`.
+2. Create the tests for the class diagram we created above.
+
+**In Small Groups**
+3. Review the tests that each member created.  Ask questions, and reach out to your instructor for help, if needed!
+
+
+### Implement Classes
+
+**Individually**
```

Comment:
I like this idea of working on the classes individually first! Looking back we do lean very heavily towards pair/group work. This can definitely be a fill the truck as we go thing, but do want to make sure students have time to think on their own.

Reply:
Agreed!